src/rust/engine/options/src/args.rs

```diff
@@ -97,20 +97,25 @@ impl Arg {
 #[derive(Debug)]
 pub struct Args {
     args: Vec<Arg>,
+    #[allow(dead_code)]
+    pub passthrough_args: Option<Vec<String>>,
 }
 
 impl Args {
     // Create an Args instance with the provided args, which must *not* include the
     // argv[0] process name.
     pub fn new<I: IntoIterator<Item = String>>(arg_strs: I) -> Self {
         let mut args: Vec<Arg> = vec![];
+        let mut passthrough_args: Option<Vec<String>> = None;
         let mut scope = Scope::Global;
-        for arg_str in arg_strs.into_iter() {
+        let mut args_iter = arg_strs.into_iter();
+        while let Some(arg_str) = args_iter.next() {
             if arg_str.starts_with("--") {
                 let mut components = arg_str.splitn(2, '=');
                 let flag = components.next().unwrap();
-                if flag.is_empty() {
+                if flag.len() == 2 {
                     // We've hit the passthrough args delimiter (`--`), so don't look further.
+                    passthrough_args = Some(args_iter.collect::<Vec<String>>());
```

Comment:
Wouldn't it be clearer to split these cases? If the `arg_str` is *exactly* `--` we collect the remainder and break, otherwise it's a flag. Right now the components is a bit odd since we *know* post-fact it didn't split, because then this if is pointless.

Reply:
Yeah, that makes sense. Will change.